hkm/templatetags/hkm_tags.py

```diff
@@ -89,3 +89,41 @@ def front_page_url(collection):
 def showcase_collections(showcase):
     albums = showcase.albums.select_related('owner').all().order_by('created')
     return albums
+
+
+@register.filter
+def search_keywords(url_params):
+    keywords = []
+    banned = ["date_from", "date_to", "page"]
+
+    for key, value in url_params.items():
+        if key in banned or len(value) == 0:
+            continue
+
+        if isinstance(value, list):
+            for item in value:
+                keywords.append({"value": item, "facet_type": key})
+        else:
+            keywords.append({"value": value, "facet_type": key})
+
+    # Manually check date_from and date_to to combine them into one keyword
+    date_from = url_params.get('date_from', '')
+    date_to = url_params.get('date_to', '')
+
+    if date_from or date_to:
+        combined = "%s - %s" % (date_from, date_to)
+        keywords.append({"value": combined, "facet_type": "date_range"})
+
+    return keywords
+
+
+@register.filter()
+def return_link(url_params):
+    cleaned_params = {}
+
+    for key, value in url_params.items():
+        if value and key != "page":
```

Comment:
Now that I think of this, removing the page parameter might affect the result page user is returning to. I'll double check this.